src/EventListener/FrontendListener.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of terminal42/contao-inserttags.
+ *
+ * (c) terminal42
+ *
+ * @license MIT
+ */
+
+namespace Terminal42\InsertTagsBundle\EventListener;
+
+use Contao\CoreBundle\InsertTag\InsertTagParser;
+use Contao\CoreBundle\ServiceAnnotation\Hook;
+use Contao\CoreBundle\String\SimpleTokenParser;
+use Doctrine\DBAL\Connection;
+use Terminal42\InsertTagsBundle\InsertTagHandler;
+
+class FrontendListener
+{
+    public function __construct(private Connection $connection, private InsertTagHandler $handler, private InsertTagParser $insertTagParser, private SimpleTokenParser $simpleTokenParser) {}
+
+    /**
+     * @Hook("replaceInsertTags")
+     */
+    public function onReplaceInsertTags(string $tag)
+    {
+        $chunks = explode('::', $tag);
+
+        if ('custom' !== $chunks[0]) {
+            return false;
+        }
+
+        $records = $this->connection->fetchAllAssociative('SELECT * FROM tl_inserttags WHERE tag=? ORDER BY sorting', [$chunks[1]]);
+
+        foreach ($records as $record) {
+            if (!$this->handler->isTagValid($record)) {
+                continue;
+            }
+
+            return $this->simpleTokenParser->parse($this->insertTagParser->replaceInline($record['replacement']), $chunks);
```

Comment:
I think the fact that you can pass on arbitrary tokens is an undocumented feature. We should provide examples in the README.
Also, I think we should extend the feature a bit more by merging `$chunks` with additional tokens that make sense.
I can think of e.g. 

- `##page_language##` that contains the page language 
- `##page_id##` ...
- maybe just everything that comes from `$page->row()`? Wdyt?


I know we can restrict to pages but this means we cannot have the `else` logic. So you can only limit an insert tag to be displayed if the conditions are met but not if they're not.
I know you can just not limit the pages at all and use the `{{iflng` insert tags here too but having simple tokens here would allow for even more powerful features like this:

```
{if page_language in ["de_DE", "pl_PL"]}This{else}that{endif}
```

or

```
{if page_language matches "^de"}German{else}Not German{endif}
```

to match all German dialects 😎 

The filter for pages and groups would then just be UX improvements for the simple cases so you don't have to write

```
{if page_id in [42, 45....]}Replacement{endif}
```